### PR TITLE
GDB-7711 add result formatting of table plugin

### PR DESCRIPTION
**Pull request by boyan-tonchev**
GDB-7711: Extend table renderer plugin

What?
Resources in table with results are formatted as short uri. Also, there is a copy resource link button next to every resource in the table.
Why?
In WB has such functionality.
How?
Original table plugin is disabled. Added a new plugin "extended-table" which shows the short uri and copy buttons, it is configured as default plugin.

### CODE REVIEW

File: ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.scss [hunk 1]
```diff
@@ -23,11 +23,6 @@
         flex-direction: column;
         align-items: center;
 
-        // There is no other way to hide it but to patch the yasqe
-        .yasqe_share {
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.scss:27]: Why did you remove this? How did you replaced it?

File: ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.tsx [hunk 2]
```diff
@@ -422,6 +432,36 @@ export class OntotextYasguiWebComponent {
...
+    this.showCopyResourceLinkDialog = false;
+    this.copiedResourceLink = undefined;
+  }
+
+  /**
+   * Handler for the event fired when the copy link dialog is closed when the link is copied to the clipboard.
+   */
+  @Listen('internalResourceLinkCopiedEvent')
+  resourceLinkCopiedHandler() {
+    this.notify.emit(new NotificationMessage(NotificationMessageType.SUCCESS, this.translationService.translate('yasqe.share.copy_link.dialog.copy.message.success')));
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.tsx:449]: I'm a bit hesitant about this. Time will tell if this will survive in favor of the general purpose event bus. But if it does, then maybe we need some message type based on which the client can deside if he wants to show it or not.

File: ontotext-yasgui-web-component/src/i18n/locale-fr.json [hunk 3]
```diff
@@ -104,6 +105,9 @@
   "yasqe.actions.include_inferred.false.button.tooltip": "Inclure les données déduites dans les résultats: DÉSACTIVÉ",
   "yasqe.actions.expand_results_sameas.true.button.tooltip": "Développer les résultats sur owl:sameAs: ON",
   "yasqe.actions.expand_results_sameas.false.button.tooltip": "Développer les résultats sur owl:sameAs: OFF",
-  "yasqe.share.query.dialog.title": "Copier l'URL dans le presse-papiers",
-  "yasqe.share.query.dialog.copy.btn.label": "Copier dans le presse-papiers"
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/i18n/locale-fr.json:108]: Use the french labels

File: ontotext-yasgui-web-component/src/services/yasr/yasr-service.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+  static getPluginsConfigurations(): Map<string, any> {
+    const pluginsConfigurations = new Map<string, any>();
+    pluginsConfigurations.set('extended_table', YasrService.getExtendedTableConfiguration());
+    return pluginsConfigurations;
+  }
+
+  private static getExtendedTableConfiguration() {
+    return {
+      getCellContent: YasrService.getCellContent.bind(this)
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/yasr/yasr-service.ts:30]: To be honest, I'd prefer these plugin implementations to be in their own classes and files and not to mess with our services. You can leave it this way now, but we should consider extracting them.

**Review by svilenvelikov — CHANGES_REQUESTED** [review]

File: Yasgui/packages/yasr/src/plugins/extended-table/extended-table.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  private readonly getCellContentCustom?: (
+    binding: Parser.BindingValue,
+    prefixes?: { [label: string]: string }
+  ) => string;
+
+  constructor(yasr: Yasr) {
+    super(yasr);
+    if (yasr.config.externalPluginsConfigurations) {
+      const pluginConfiguration = yasr.config.externalPluginsConfigurations.get("extended_table");
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply Yasgui/packages/yasr/src/plugins/extended-table/extended-table.ts:17]: We decided to extend original table plugin instead to change source code.
This is ontotext implementation of table plugin, but we decided to not use ontotext in names.
For now, we override "getCellContent" function only, but this can be change in the future.

File: ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.scss (hunk 1 above)
Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.scss:27]: This is a mistake, when I merge the conflicts after rebasing. I will revert it.

Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.scss]: Done

File: ontotext-yasgui-web-component/src/components/copy-link-dialog/copy-link-dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+import {DialogConfig} from "../ontotext-dialog-web-component/ontotext-dialog-web-component";
+import {ServiceFactory} from "../../services/service-factory";
+import {TranslationService} from "../../services/translation.service";
+
+export type CopyLinkDialogConfig = {
+  dialogTitle?: string;
+  copyLink: string;
+}
+
+export interface CopyLInkObserver {
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/components/copy-link-dialog/copy-link-dialog.tsx]: Done

**Review by svilenvelikov — APPROVED** [review]
